test/File/RenameUploadTest.php

```diff
@@ -412,4 +412,23 @@ public function testReturnUnfiltered($input)
 
         $this->assertEquals($input, $filter($input));
     }
+
+    /**
+     * @return void
+     */
+    public function testTargetSameAsSource()
+    {
+        $filter = new RenameUploadMock();
+
+        $this->assertEquals(
+            [
+                'tmp_name' => $this->sourceFile,
+                'name' => $this->targetFile,
+            ],
+            $filter([
+                'tmp_name' => $this->sourceFile,
+                'name' => $this->targetFile,
```

Comment:
*WARNING*: this test will fail when BC Break https://github.com/zendframework/zend-filter/pull/79 fix will revert `tmp_name/name` behavior